### ISSUE TITLE
Support parallel processing in dump file parsing

### ISSUE THREAD

**Issue by mkroetzsch** [Wikidata/Wikidata-Toolkit#39]
All stages of the dump file processing should make use of parallel processing to improve performance. To enable this, these objects need to get access to a shared Executor object that manages the threads. The following tasks should be executed in independent threads:
- downloading required dump files (in parallel to processing the first ones)
- unzipping the dump files
- parsing the XML contents
- parsing the JSON for a revision

For some of these, separate issues should be created later.

**Comment by brett-matson** [comment]
Hi Guys, 

Sorry, I'm not sure what the etiquette is for commenting on closed issues, but I wonder whether this one would benefit from some more thought. 

When parsing the dump file, I only ever get one CPU core at a time doing work (see the image below). It appears that the parsing is CPU bound for a single core, so potentially the Toolkit could gain performance by using multiple cores. Can anyone confirm whether Jackson is capable of parallel parsing, and if so, whether that could be enabled in the Toolkit?

![image](https://user-images.githubusercontent.com/74486738/107116597-8344bc80-68c8-11eb-96e4-829bae4c0d58.png)

**Comment by wetneb** [comment]
@brett-matson perhaps you could just open a new issue, requesting this improvement? Personally I don't use WDTK on dumps so I don't have experience with this.

**Comment by thadguidry** [comment]
@brett-matson Depending on your exact needs, you might also take a look at other clients that support processing Wikidata dumps such as [Knowledge Graph Toolkit (Python)](https://github.com/usc-isi-i2/kgtk)
